From https://bugs.gentoo.org/723374
From: Tiernan Hubble
Date: Mon, 23 Aug 2021 12:17:00 -0700
Subject: [PATCH] Fix CMake detection of GCC 10

from Pierre Willenbrock:

vtk uses regular expressions to detect compiler version which they must be updating
every time the gcc major version goes up. In this case, the regular expression in 
CMake/VTKGenerateExportHeader.cmake([3-9]\\.[0-9]\\.[0-9]*) fails to match "10.2.1".

That fails to properly set COMPILER_HAS_HIDDEN_INLINE_VISIBILITY when needed, but 
seemingly only for the first run. This causes the visibility attributes to be missing
on the function declarations, because the #defines for that are left empty. For some
reason, COMPILER_HAS_HIDDEN_INLINE_VISIBILITY does get set later, leading 
to -fvisibility=hidden to be passed to the compiler, making all functions hidden, thus
not exported from the shared object. This also gets "fixed" by rerunning cmake in the
build directory.

this patch fixes the version regex to detect GCC versions with a 2-digit major version

--- a/CMake/VTKGenerateExportHeader.cmake
+++ b/CMake/VTKGenerateExportHeader.cmake
@@ -174,8 +174,12 @@ macro(_vtk_test_compiler_hidden_visibility)
     execute_process(COMMAND ${CMAKE_C_COMPILER} --version
       OUTPUT_VARIABLE _gcc_version_info
       ERROR_VARIABLE _gcc_version_info)
-    string(REGEX MATCH "[3-9]\\.[0-9]\\.[0-9]*"
+    string(REGEX MATCH "[1-9][0-9]\\.[0-9]\\.[0-9]*"
       _gcc_version "${_gcc_version_info}")
+    if(NOT _gcc_version)
+      string(REGEX MATCH "[3-9]\\.[0-9]\\.[0-9]*"
+        _gcc_version "${_gcc_version_info}")
+    endif()
     # gcc on mac just reports: "gcc (GCC) 3.3 20030304 ..." without the
     # patch level, handle this here:
     if(NOT _gcc_version)
